### PR TITLE
Add run_android_tests workflow

### PR DESCRIPTION
**Pull request by kit1980**
Android emulator tests workflow on Linux Github hosted runners, for now triggers on master/main/release push.

Example successful run: https://github.com/pytorch/pytorch/runs/5488927387

### CODE REVIEW

File: .github/workflows/run_android_tests.yml [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+  push:
+
+defaults:
+  run:
+    shell: bash -e -l {0}
+
+jobs:
+
+  build:
+    runs-on: macos-11
```
Comment (janeyx99) [inline .github/workflows/run_android_tests.yml:15]: We are limited on mac runners--is it possible to run these on Linux runners?

**Comment by kit1980** [comment]
> Do you have a link to a successful run?

https://github.com/pytorch/pytorch/runs/5354154503

File: .github/workflows/run_android_tests.yml (hunk 1 above)
Comment (janeyx99) [reply .github/workflows/run_android_tests.yml:15]: Yes, this is true I see the build is 55min while the test is 3min

File: .github/workflows/run_android_tests.yml [hunk 2]
```diff
@@ -0,0 +1,60 @@
+name: android-tests
+
+on:
+  push:
+    branches:
+      - master
+      - main
+      - release/*
```
Comment (malfet) [inline .github/workflows/run_android_tests.yml:8]: Please make sure to schedule the workflow on release and ciflow tags
```suggestion
      - release/*
    tags:
      # NOTE: Binary build pipelines should only get triggered on release candidate builds
      # Release candidate tags look like: v1.11.0-rc1
      - v[0-9]+.[0-9]+.[0-9]+-rc[0-9]+
    - 'ciflow/trunk/*'
    - 'ciflow/android/*'
```

**Review by malfet — APPROVED** [review]
LGTM, but can you please elaborate what have changed from failed run that we've observed yesterday to this one?

**Comment by kit1980** [comment]
> LGTM, but can you please elaborate what have changed from failed run that we've observed yesterday to this one?

The change is that older api-level: 25 is needed - that's what the test requires. Later we can update the test and the API level.

**Pull request closed without merging**